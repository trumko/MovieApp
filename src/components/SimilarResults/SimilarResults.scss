.mr_moviePreviews {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}
