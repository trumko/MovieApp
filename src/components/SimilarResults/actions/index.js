import { SIMILAR_RESULTS_GET, SIMILAR_RESULTS_CLEAR } from "./types";

export function getSimilarResults(payload) {
  return { type: SIMILAR_RESULTS_GET, payload };
}

export function clearSimilarResults(payload) {
  return { type: SIMILAR_RESULTS_CLEAR, payload };
}
