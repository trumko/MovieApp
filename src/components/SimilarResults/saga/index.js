import { takeEvery, call, put } from "redux-saga/effects";
import { fetchSimilarResults } from 'utils/api';
import {
  SIMILAR_RESULTS_GET, SIMILAR_RESULTS_GET_SUCCESS, SIMILAR_RESULTS_GET_FAIL
} from "../actions/types";

function* getSimilarResults(action) {
  try {
    const payload = yield call(fetchSimilarResults, action.payload);
    yield put({ type: SIMILAR_RESULTS_GET_SUCCESS, payload });
  } catch (e) {
    yield put({ type: SIMILAR_RESULTS_GET_FAIL, payload: e });
  }
}

export default function* similarResultsSaga() {
  yield takeEvery(SIMILAR_RESULTS_GET, getSimilarResults);
}
