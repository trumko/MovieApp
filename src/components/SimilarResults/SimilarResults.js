import React, { Component } from 'react';
import { withRouter } from 'react-router-dom'
import { connect } from "react-redux";

import MoviePreview from 'components/MoviePreview/MoviePreview'
import { getSimilarResults, clearSimilarResults } from './actions'

import './SimilarResults.scss';

export class SimilarResults extends Component {
  componentDidMount() {
    console.log(1);
    if (this.props.movie && !this.props.similarResults === null) {
      this.props.getSimilarResults(this.props.movie.genres);
    }
  }

  componentDidUpdate() {
    console.log(2);

    if (this.props.movie && this.props.similarResults === null) {
      console.log(2.1);
      this.props.getSimilarResults(this.props.movie.genres);
    }
  }

  render() {
    const { similarResults } = this.props;

    if (!similarResults) {
      return null;
    }

    return (
      <div className="mr_moviePreviews">
        {similarResults.map(({ title, release_date, genres, poster_path, id }) => {
          return (
            <MoviePreview
              title={title}
              date={release_date}
              genres={genres}
              posterUrl={poster_path}
              key={id}
              id={id}
            />
          );
        })}
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  getSimilarResults: params => dispatch(getSimilarResults(params)),
  clearSimilarResults: () => dispatch(clearSimilarResults())
})

const mapStateToProps = (state) => ({
  movie: state.movieReducer.movie,
  similarResults: state.similarResultsReducer.similarResults
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SimilarResults))
