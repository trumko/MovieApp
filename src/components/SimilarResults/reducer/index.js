import { SIMILAR_RESULTS_GET_SUCCESS, SIMILAR_RESULTS_GET_FAIL, SIMILAR_RESULTS_CLEAR } from "../actions/types";

const initialState = {
  similarResults: null,
};

export default function movieReducer(state = initialState, action) {
  switch(action.type) {
    case SIMILAR_RESULTS_GET_SUCCESS:
      return {...state, similarResults: action.payload.data}

    case SIMILAR_RESULTS_GET_FAIL:
    case SIMILAR_RESULTS_CLEAR:
      return {...state, similarResults: null}

    default:
      break;
  }

  return state;
}
