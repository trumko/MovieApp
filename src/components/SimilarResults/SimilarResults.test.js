import React from 'react'
import { shallow } from 'enzyme'
import { SimilarResults } from './SimilarResults'

const mockedMovies = [
  {
    "id": 308266,
    "title": "War Dogs",
    "tagline": "An American dream",
    "vote_average": 6.8,
    "vote_count": 1773,
    "release_date": "2016-08-17",
    "poster_path": "https://image.tmdb.org/t/p/w500/wJXku1YhMKeuzYNEHux7XtaYPsE.jpg",
    "overview": "Based on the true story of two young men, David Packouz and Efraim Diveroli, who won a $300 million contract from the Pentagon to arm America's allies in Afghanistan.",
    "budget": 40000000,
    "revenue": 86234523,
    "genres": [
      "Comedy",
      "Crime",
      "Drama"
    ],
    "runtime": 114
  },
  {
    "id": 553,
    "title": "Dogville",
    "tagline": "A quiet little town not far from here.",
    "vote_average": 7.7,
    "vote_count": 823,
    "release_date": "2003-05-19",
    "poster_path": "https://image.tmdb.org/t/p/w500/g1xLrof2RVgtHBB4fLvR5Xr8l5x.jpg",
    "overview": "A barren soundstage is stylishly utilized to create a minimalist small-town setting in which a mysterious woman named Grace hides from the criminals who pursue her. The town is two-faced and offers to harbor Grace as long as she can make it worth their effort, so Grace works hard under the employ of various townspeople to win their favor. Tensions flare, however, and Grace's status as a helpless outsider provokes vicious contempt and abuse from the citizens of Dogville.",
    "budget": 10000000,
    "revenue": 16680836,
    "genres": [
      "Crime",
      "Drama",
      "Thriller"
    ],
    "runtime": 178
  }
]

describe('SimilarResults', () => {
  it('SimilarResults without Movies should be rendered correctly', () => {
    const component = shallow(
      <SimilarResults similarResults={undefined} />
    );

    expect(component).toMatchSnapshot();
  });

  it('SimilarResults with Movies should be rendered correctly', () => {
    const component = shallow(
      <SimilarResults similarResults={mockedMovies} />
    );

    expect(component).toMatchSnapshot();
  });
});
