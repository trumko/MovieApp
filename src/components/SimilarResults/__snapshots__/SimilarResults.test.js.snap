// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimilarResults SimilarResults with Movies should be rendered correctly 1`] = `
<MoviePreviews
  movies={
    Array [
      Object {
        "budget": 40000000,
        "genres": Array [
          "Comedy",
          "Crime",
          "Drama",
        ],
        "id": 308266,
        "overview": "Based on the true story of two young men, David Packouz and Efraim Diveroli, who won a $300 million contract from the Pentagon to arm America's allies in Afghanistan.",
        "poster_path": "https://image.tmdb.org/t/p/w500/wJXku1YhMKeuzYNEHux7XtaYPsE.jpg",
        "release_date": "2016-08-17",
        "revenue": 86234523,
        "runtime": 114,
        "tagline": "An American dream",
        "title": "War Dogs",
        "vote_average": 6.8,
        "vote_count": 1773,
      },
      Object {
        "budget": 10000000,
        "genres": Array [
          "Crime",
          "Drama",
          "Thriller",
        ],
        "id": 553,
        "overview": "A barren soundstage is stylishly utilized to create a minimalist small-town setting in which a mysterious woman named Grace hides from the criminals who pursue her. The town is two-faced and offers to harbor Grace as long as she can make it worth their effort, so Grace works hard under the employ of various townspeople to win their favor. Tensions flare, however, and Grace's status as a helpless outsider provokes vicious contempt and abuse from the citizens of Dogville.",
        "poster_path": "https://image.tmdb.org/t/p/w500/g1xLrof2RVgtHBB4fLvR5Xr8l5x.jpg",
        "release_date": "2003-05-19",
        "revenue": 16680836,
        "runtime": 178,
        "tagline": "A quiet little town not far from here.",
        "title": "Dogville",
        "vote_average": 7.7,
        "vote_count": 823,
      },
    ]
  }
/>
`;

exports[`SimilarResults SimilarResults without Movies should be rendered correctly 1`] = `""`;
